### ISSUE TITLE
Kubernetes deployment

### ISSUE THREAD

**Issue by apetrut** [skoruba/Duende.IdentityServer.Admin#140]
We've been trying to deploy Admin UI to Kubernetes.

Currently all three services (admin, admin-api and sts) are running in the same pod, The sts one is running and the other two are restarting every 6-7 seconds.

![image](https://user-images.githubusercontent.com/13006228/218141708-8a0acfef-1970-420d-9c13-e4d3d36ba33a.png)

Do you have any recommendations about running all the services in the same pod or in different pods?

Also, is it possible to use the url format: 

https://your-domain-name.com/admin 

instead of 

https://admin.your-domain-name.com?

**Comment by skoruba** [comment]
I do not know, I do not have experience with k8s.

**Comment by albrookesplowman** [comment]
I would probably deploy each part to a different k8s service (with 1 or many pod replicas) and you can then reference each element by the service name (assuming that it is all in the same cluster)

**Comment by apetrut** [comment]
@albrookesplowman thanks for your answer, we will try that approach.

Also, we are getting the error below:

![image](https://user-images.githubusercontent.com/13006228/218298475-a6ba2b7b-6356-4308-ae59-6ad36e93c2c3.png)

It seems related to timeouts, but we haven't specified any.

**Comment by albrookesplowman** [comment]
@apetrut - have you tried looking at the logs of the pod or **kubectl describe** on the pod.  There might be some clues in one of those.  These type of back-off errors are a nightmare to find... Are you deploying via kubectl or via helm?

**Comment by apetrut** [comment]
@albrookesplowman yes, of course I've checked the logs but it's not a lot of info in there:

![image](https://user-images.githubusercontent.com/13006228/218407444-39afdb52-cfa6-45b0-8438-d8fefbb79c98.png)

I am deploying using Azure DevOps with Kubernetes and using Lens Dekstop for inspecting the pods and logs.

**Comment by albrookesplowman** [comment]
@apetrut - not sure what is happening there... it will probably need log levels changing to debug to see how far through the start up process it is getting. Sorry I couldn't be more help...

**Comment by apetrut** [comment]
@albrookesplowman I created 3 different services and they are running correctly each in their own pod. I didn't get any errors so far and they are all green. The "back-off restarting" error must have been something related to running all the services in the same pod.

Now we have some DNS issues (internal and external) that will try to solve in the next couple of days. We are following the documentation provided by @bravecobra. 

Thanks for your help.

**Comment by apetrut** [comment]
Now we're getting this error:

![image](https://user-images.githubusercontent.com/13006228/220326204-0c03a010-cdc2-4bbc-ae1a-a2fc9cb1f690.png)

It seems it can't perform health check for identity server. I have checked the url and it was the correct one.
